src/basis/dom/event.js

```diff
@@ -70,33 +71,54 @@
     mousewheel: ['wheel', 'mousewheel', 'DOMMouseScroll']
   };
 
-  var DEPRECATED = /^(returnValue|keyLocation|layerX|layerY|webkitMovementX|webkitMovementY|keyIdentifier)$/;
-
-  var KEYBOARD_EVENTS = [
-    'keyup',
-    'keydown',
-    'keypress'
+  var DEPRECATED_PROPERTIES = [
+    'returnValue',
+    'keyLocation',
+    'layerX',
+    'layerY',
+    'webkitMovementX',
+    'webkitMovementY',
+    'keyIdentifier'
   ];
 
-  var MOUSE_EVENTS = [
-    'click',
-    'dblclick',
-    'mousedown',
-    'mouseup',
-    'mouseover',
-    'mousemove',
-    'mouseout',
-    'mouseenter',
-    'mouseleave'
-  ].concat(BROWSER_EVENTS.mousewheel);
-
-  function isKeyboardEvent(event) {
-    return KEYBOARD_EVENTS.indexOf(event.type) != -1;
-  }
+  var TYPE_KEYBOARD = 0;
+  var TYPE_MOUSE = 1;
+  var TYPE_TOUCH = 2;
+  var TYPE_POINTER = 3;
+
+  var INPUT_TYPE = {
+    keydown:       TYPE_KEYBOARD,
+    keypress:      TYPE_KEYBOARD,
+    keyup:         TYPE_KEYBOARD,
+
+    click:         TYPE_MOUSE,
+    dblclick:      TYPE_MOUSE,
+    mousedown:     TYPE_MOUSE,
+    mouseup:       TYPE_MOUSE,
+    mouseover:     TYPE_MOUSE,
+    mousemove:     TYPE_MOUSE,
+    mouseout:      TYPE_MOUSE,
+    mouseenter:    TYPE_MOUSE,
+    mouseleave:    TYPE_MOUSE,
+
+    touchstart:    TYPE_TOUCH,
+    touchmove:     TYPE_TOUCH,
+    touchend:      TYPE_TOUCH,
+    touchcancel:   TYPE_TOUCH,
+
+    pointerover:   TYPE_POINTER,
+    pointerenter:  TYPE_POINTER,
+    pointerdown:   TYPE_POINTER,
+    pointermove:   TYPE_POINTER,
+    pointerup:     TYPE_POINTER,
+    pointercancel: TYPE_POINTER,
+    pointerout:    TYPE_POINTER,
+    pointerleave:  TYPE_POINTER
+  };
 
-  function isMouseEvent(event) {
-    return MOUSE_EVENTS.indexOf(event.type) != -1;
-  }
+  extend(INPUT_TYPE, BROWSER_EVENTS.mousewheel.reduce(function(result, eventName){
```

Comment:
Looks a little bit complex. Maybe add those events explicit to INPUT_TYPE (with comment that those events are all about mouse wheel)?